lib/src/commerce/commerce_helpers.dart

```diff
@@ -33,4 +33,63 @@ String promotionActionTypeString(PromotionActionType productActionType) {
     case PromotionActionType.View:
       return "view";
   }
-}
\ No newline at end of file
+}
+
+// Web integer values for product action types differ from Dart Enums
+// https://github.com/mParticle/mparticle-web-sdk/blob/master/src/types.js#L255-L267
+int? getWebSDKProductActionType(ProductActionType productActionType) {
+  const jsProductActionTypes = {
+    'AddToCart': 1,
+    'RemoveFromCart': 2,
+    'Checkout': 3,
+    'CheckoutOption': 4,
+    'Click': 5,
+    'ViewDetail': 6,
+    'Purchase': 7,
+    'Refund': 8,
+    'AddToWishlist': 9,
+    'RemoveFromWishlist': 10,
+  };
+
+  switch (productActionType) {
+    case ProductActionType.AddToCart:
+      return jsProductActionTypes['AddToCart'];
+    case ProductActionType.RemoveFromCart:
+      return jsProductActionTypes['RemoveFromCart'];
+    case ProductActionType.AddToWishList:
+      return jsProductActionTypes['AddToWishlist'];
+    case ProductActionType.RemoveFromWishlist:
+      return jsProductActionTypes['RemoveFromWishlist'];
+    case ProductActionType.Checkout:
+      return jsProductActionTypes['Checkout'];
+    case ProductActionType.Click:
+      return jsProductActionTypes['CheckoutOption'];
+    case ProductActionType.ViewDetail:
+      return jsProductActionTypes['Click'];
+    case ProductActionType.Purchase:
+      return jsProductActionTypes['ViewDetail'];
+    case ProductActionType.Refund:
+      return jsProductActionTypes['Purchase'];
+    case ProductActionType.CheckoutOptions:
+      return jsProductActionTypes['Refund'];
+  }
+}
+
+// Web integer values for promotion action types differ from Dart Enums
+// https://github.com/mParticle/mparticle-web-sdk/blob/master/src/types.js#L324-L328
+int? getWebSDKPromotionActionType(PromotionActionType promotionActionTypeEnum) {
+  const jsPromotionActionTypes = {
+    'Unknown': 0,
```

Comment:
what is the use for 'Unknown'? Seems like it prints an error on 92 rather than ever using it

Reply:
It appears web and the server both support `Unknown`, but iOS/Android don't. I'll log it as an inconsistency, and remove it from here for now since the Dart layer doesn't have it.